Neuro/Minigames/Completion/MinigameSolver.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using UnityEngine;
+
+namespace Neuro.Minigames.Completion;
+
+public abstract class MinigameSolver
+{
+    static MinigameSolver()
+    {
+        MinigameSolvers = Assembly.GetExecutingAssembly().GetTypes()
+            .Where(t => t.GetCustomAttribute<MinigameSolverAttribute>() is { })
+            .Where(t => t.IsAssignableTo(typeof(MinigameSolver)))
+            .Select(solverType => (solverType.GetCustomAttribute<MinigameSolverAttribute>()!.TaskTypes, solverType))
+            .SelectMany(t => t.TaskTypes.Select(taskType => (taskType, t.solverType)))
+            .ToDictionary(t => t.taskType, t => t.solverType);
```

Comment:
Do we really need reflection for this? I don't think just initializing a dictionary here manually would be that bad.

Reply:
Reflection is fine since it's being called in a static constructor, and I wouldn't want to manually initialize a dictionary with like 100 elements.